File: leetcode/leetcode-53.py
from typing import List, NamedTuple

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans = -1111111111111
        summ = 0
        for v in nums:
            summ += v
            ans = max(ans, summ)
            if summ < 0:
                summ = 0
        return ans

if __name__ == "__main__":
    so = Solution()
    print(so.maxSubArray([-2,1,-3,4,-1,2,1,-5,4]))

File: leetcode/leetcode-53.cpp
/*
 * @File Name          :leetcode-53.cpp
 * @Author             :cc
 * @Version            :1.0.0
 * @Date               :2022/1/5 11:17
 * @Description        :
 * @Function List      :
 * @History            :
 */

#include "vector"
#include "climits"
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = INT_MIN;
        int sum = 0;
        for(auto &x: nums){
            sum += x;
            ans = max(ans, sum);
            sum = max(sum, 0);
        }return ans;
    }
};